File: Namrata-Sanger/848. Shifting Letters.js
/**
 * @param {string} s
 * @param {number[]} shifts
 * @return {string}
 */
 /*
 Day 18
 848. Shifting Letters
 Link: https://leetcode.com/problems/shifting-letters/
 Level: Medium
 */
var shiftingLetters = function(s, shifts) {
    let sum = 0;
    for(let i = shifts.length-1; i>=0; i--){
        sum += shifts[i]%26;
        shifts[i] = String.fromCharCode((sum+s.charCodeAt(i) - 97) % 26 + 97);
    }
    return shifts.join('');
    
};

File: Dhairyashil-Shinde/848-shifting-letters.cpp
class Solution {
public:
    
    string shiftingLetters(string s, vector<int>& shifts) {
        
        int n = s.length();
        long long int sum = 0;
        
        for(int i = n-1;i>=0;i--){
            
            sum = (sum + shifts[i])%26 ; 
            
            long long int temp = s[i]-'a';
            
            temp = (temp + sum)%26;
            
            s[i] = 'a' + temp;
            
        }
        return s;
    }
};

/*
    string shiftingLetters(string s, vector<int>& shifts) {
        long shift = 0;
        for(int i=s.size()-1; i>=0; i--) {
            s[i] = ( (s[i]-'a') + (shift+shifts[i]) % 26) % 26 +'a';
            shift += shifts[i];
        }
        return s;
    }
*/
